### ISSUE TITLE
Screenshots are sometimes blank

### ISSUE THREAD

**Issue by wwilk** [ui4j/ui4j#66]
I'm trying to take a screenshot of multiple websites concurrently, and some of them are blank when saved to file. The code I'm using:

```java
    private static final String SCREENSHOT_DIRECTORY = "/Users/wilk/websites/images/before";
	
	public static void main(String[] args){
		System.setProperty("ui4j.headless", "true");
		
		ExecutorService service = Executors.newFixedThreadPool(20);
		
		for(int i=1;i<=100;i++){
    		service.execute(new Task(BrowserFactory.getWebKit(), i));
    	}
	}

	private static class Task implements Runnable{
		private final BrowserEngine engine;
		private final int index;

		public Task(BrowserEngine engine, int index) {
			this.engine = engine;
			this.index = index;
		}

		public void run(){
			try{
				doRun(engine);
			} catch(Exception e){
				throw new RuntimeException();
			}
		}
		
		public void doRun(BrowserEngine engine) throws FileNotFoundException {
			String url = "http://www.ebay.com/sch/Nonfiction-/171243/i.html?_pgn=" + index;

			Page page = engine.navigate(url);
    		page.show(true);
    	
    		page.captureScreen(new FileOutputStream(new File(SCREENSHOT_DIRECTORY + index + ".png")));
		}
	}
```

Does anyone know what may be the reason of that? I tried waiting with CountDownLatch for document to be loaded (countDown() in [Interceptor#afterLoad()](https://github.com/ui4j/ui4j/blob/master/ui4j-api/src/main/java/com/ui4j/api/interceptor/Interceptor.java) and await() before taking screenshot), but that didn't help. 

It seems to happen more often if I don't use singleton instance of [WebkitBrowser](https://github.com/ui4j/ui4j/blob/master/ui4j-webkit/src/main/java/com/ui4j/webkit/WebKitBrowser.java), passing new instance to every task instead (created with ServiceProvider). BTW are there any other issues with this approach? It would be good to have multiple instances, for performance reasons, since WebkitBrowser#navigate is synchronised and blocks for significant amount of time.

Also when I wait some time after taking a screenshot and take another one of the same Page instance, then the second one is usually blank, despite the fact that often the first one isn't.

**Comment by jgangemi** [comment]
i see the same behavior if i run the code from `examples`. 

are you doing this in headless mode? i'm using the monocle library and i think part of the issue is the program is able to exit before the underlying framebuffer process starts. i need to try this using xvfb to see if i get the same results.

**Comment by wwilk** [comment]
Yes, I was doing that in either headless with monocle or headful mode. If you could share your observations about headless mode based on xvfb, I would be really interested in seeing them.

**Comment by ui4j** [comment]
Hi,
it seems there is a concurrency problem for the Page.captureScreen() method.
I will try to solve the problem.

**Comment by kyeljmd** [comment]
Do we have any updates on this issue?
